packages/icalendar/type/Recur.pkl

```diff
@@ -0,0 +1,198 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2025 Apple Inc. and the Pkl project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+/// A recurrence rule.
+///
+/// https://datatracker.ietf.org/doc/html/rfc5545#section-3.3.10
+module icalendar.Recur
```

Comment:
This and everywhere else: move into the root package dir (should be packages/icalendar/Recur.pkl)

Reply:
I was hoping to separate out the iCalendar data types (DATE, TIME, DATE-TIME, RECUR, etc.) from the (future) calendar object or components types (VCALENDAR, VEVENT, VTODO, etc.) by directory without having to cut a major release down the road.

Reply:
Makes sense!